plugin/pkg/shoot/validator/admission.go

```diff
@@ -148,30 +148,28 @@ func (v *ValidateShoot) Admit(ctx context.Context, a admission.Attributes, o adm
 		return nil
 	}
 
-	// Ignore updates if shoot spec hasn't changed
+	shoot, ok := a.GetObject().(*core.Shoot)
+	if !ok {
+		return apierrors.NewInternalError(errors.New("could not convert resource into Shoot object"))
+	}
+
+	// We only want to validate fields in the Shoot against the CloudProfile/Seed constraints which have changed.
+	// On CREATE operations we just use an empty Shoot object, forcing the validator functions to always validate.
+	// On UPDATE operations we fetch the current Shoot object.
+	var oldShoot = &core.Shoot{}
+
+	// Exit early if shoot spec hasn't changed
 	if a.GetOperation() == admission.Update {
-		newShoot, ok := a.GetObject().(*core.Shoot)
-		if !ok {
-			return apierrors.NewInternalError(errors.New("could not convert resource into Shoot object"))
-		}
-		oldShoot, ok := a.GetOldObject().(*core.Shoot)
+		old, ok := a.GetOldObject().(*core.Shoot)
 		if !ok {
 			return apierrors.NewInternalError(errors.New("could not convert old resource into Shoot object"))
 		}
+		oldShoot = old
 
 		// do not ignore metadata updates to detect and prevent removal of the gardener finalizer or unwanted changes to annotations
-		if reflect.DeepEqual(newShoot.Spec, oldShoot.Spec) && reflect.DeepEqual(newShoot.ObjectMeta, oldShoot.ObjectMeta) {
+		if reflect.DeepEqual(shoot.Spec, oldShoot.Spec) && reflect.DeepEqual(shoot.ObjectMeta, oldShoot.ObjectMeta) {
 			return nil
 		}
-
-		if newShoot.Spec.Provider.Type != oldShoot.Spec.Provider.Type {
```

Comment:
Can you shortly explain why this check is removed?

Reply:
It is already validated in the static validation (`pkg/core/validation)` and isn't necessary to be validated here in the dynamic validation again.